PocketKit/Sources/Analytics/AppEvents/Widget.swift

```diff
@@ -0,0 +1,40 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+
+public extension Events {
+    struct Widget {}
+}
+
+public extension Events.Widget {
+    struct Tags { }
+}
+
+public extension Events.Widget {
+    /// Fired when a user taps on a widget and it opens the application
+    static func widgetOpenEngagement() -> Event {
+        return Engagement(
+            .general,
+            uiEntity: UiEntity(
+                .screen,
+                identifier: "widget-extension.tapped"
+            )
+        )
+    }
+
+    /// TODO: not necessarily an engagement event, how should we categorize this
+    /// Fired when a user launches the applications and provides details of the widget
+    static func widgetDetailsEngagement(details: [String]) -> Event {
+        return Engagement(
+            .general,
+            uiEntity: UiEntity(
+                .screen,
+                identifier: "widget-extension.details",
+                componentDetail: String(details.joined(separator: ", ")),
+                value: String(details.count)
+            )
+        )
+    }
```

Comment:
not sure if this is the best representation of tracking this type of detail @Gio2018. This also only occurs at launch and not necessarily when a user installs / uninstalls an application. We can use this as a starting point and track the difference if the user's userDefaults value has changed based on whether they have more or less widgets on the next launch.